File: src/main/java/jk/kamoru/ground/info/domain/Studio.java
package jk.kamoru.ground.info.domain;

import java.net.URL;
import java.util.Date;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
public class Studio implements Info<String> {

  @NotBlank
  String name;
  String company;
  URL homepage;
  long lastModified;

  public Studio(String key) {
    setKey(key);
    this.company = "";
    this.homepage = null;
    this.lastModified = -1;
  }

  @Override
  public String getKey() {
    return name;
  }

  @Override
  public void setKey(String key) {
    this.name = key;
  }

  @Override
  public void touch() {
    lastModified = new Date().getTime();
  }

}

File: flay-web/backend/src/domain/studio.ts
/** Studio 도메인 (Info<String> 대응) */
export interface Studio {
  name: string;
  company: string;
  /** 홈페이지 URL 문자열 (Java URL -> string) */
  homepage: string;
  lastModified: number;
}

/**
 * 새 Studio 생성
 * @param name 스튜디오 이름 (key)
 */
export function createStudio(name: string): Studio {
  return { name, company: '', homepage: '', lastModified: -1 };
}
